audio-worklet/src/bitcrusher.js

```diff
@@ -0,0 +1,135 @@
+distribution.
+    * Neither the name of Google Inc. nor the names of its
+contributors may be used to endorse or promote products derived from
+this software without specific prior written permission.
+
+THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+"AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+(INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+*/
+
+/** 
+ * Bitcrusher implements an AudioNode-like wrapper for a ScriptProcessor object 
+ * that reduces the sample rate and bit depth of an audiobuffer.
+ * @param {BaseAudioContext} context the audio context
+ * @param {Number} options.bufferSize holds the size of an onaudioprocess window
+ * @param {Number} options.inputChannels holds the number of input channels
+ * @param {Number} options.outputChannels holds the number of output channels
+ * @param {Number} options.bitDepth holds the number of bits for each sample
+ * @param {Number} options.reduction holds the degree of reduction
+*/
+class Bitcrusher {
+  
+  constructor (context, options) {
+    this.context_ = context;
+    this.bitDepth = options.bitDepth; 
+    this.reduction = options.reduction;
+    this.node_ = this.context_.createScriptProcessor(options.buffersize, 
+      options.inputChannels, options.outputChannels);
+
+    // bind the AudioNode onaudioproess method to the function defined below 
+    this.node_.onaudioprocess = this.onaudioprocess_.bind(this); 
+  }
+  
+  /**
+    Bit crush upon receiving input audio signal, applying two signal distortion 
+    effects to the output buffer controlled by this.bitDepth and this.reduciton
+    @param {AudioProcessingEvent} event holds input and output buffers
+  */
+  onaudioprocess_ (event) {
+    for (let i = 0; i < event.inputBuffer.numberOfChannels; i++) {
+      Bitcrusher.bitcrush(this.reduction, this.bitDepth, 
+        event.inputBuffer.getChannelData(i), 
+        event.outputBuffer.getChannelData(i));
+    }
+  }
+
+  /**
+    Reduce the information in given inputBuffer with specified factor and 
+    precision, placing results in oututBuffer.
+    @param {Number} factor to reduce by
+    @param {Number} precision or bit depth of post-processed samples
+    @param {Float32Array} inputBuffer pre-processed buffer
+    @param {Float32Array} outputBuffer post-processed buffer
+  */
+  static bitcrush(factor, precision, inputBuffer, outputBuffer) {
```

Comment:
So if this function gets called like this:

```js
Bitcrusher.bitcrush(...)
```

Isn't it repetitive? How about `processBuffers()`? We commonly used the pattern of `verbNoun()` when possible and I think it is a very good practice in general.